pkg/helpers/helpers.go

```diff
@@ -14,3 +14,20 @@ func DatesEqual(d1, d2 *time.Time) bool {
 	daysSame := d1.Day() == d2.Day()
 	return yearsSame && monthsSame && daysSame
 }
+
+func MapSlice[T any, R any](s []T, fx func(s T) R) []R {
+	var r []R
+	for _, item := range s {
+		r = append(r, fx(item))
+	}
+	return r
+}
+
+func ToMap[T any, K comparable](s []T, fx func(s T) K) map[K]T {
+	r := map[K]T{}
+	for _, item := range s {
+		key := fx(item)
+		r[key] = item
+	}
+	return r
+}
```

Comment:
These are nice generic helper functions!

FWIW, samber/lo also has a lot of stuff akin to this if you ever end up wanting to lean on that package for things of this nature!

- [lo.MapToSlice](https://github.com/samber/lo?tab=readme-ov-file#maptoslice)
- [lo.SliceToMap (AKA lo.Associate)](https://github.com/samber/lo?tab=readme-ov-file#associate-alias-slicetomap)

Reply:
If we keep the ones written here, though, I'd ask that we put some comments on the function so it's clear what their purpose is (since these generic functions can be a bit tough to read out of context!)

Case in point: I thought the MapSlice was converting a Map to a Slice, but it's actually Mapping a Slice (and would more appropriately be replaced by [lo.Map](https://github.com/samber/lo?tab=readme-ov-file#map))

Reply:
I didn't realize we already had something like lodash in the project. Should I just use those or do you think we should keep the helpers?